adagrams/game.py

```diff
@@ -1,11 +1,154 @@
+import random 
+import pdb
+
 def draw_letters():
-    pass
+
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    # create a list that will eventually be filled with 10 different letters
+    LENGTH_OF_HAND = 10
+    hand = []
+    
+    # based on the desired length of the hand (in this case, 10), 
+    # the loop will simulate a player 'picking' 10 random letters 
+    # for their hand
+    
+    while len(hand) < LENGTH_OF_HAND: 
+        # random letter generator (upper case)
+        random_letter = chr(random.randint(ord('A'), ord('Z')))
+
+        if LETTER_POOL[random_letter] == 0: 
+            continue
+        
+        else: 
+            LETTER_POOL[random_letter] -= 1
+            hand.append(random_letter)
+
+    return hand 
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # check each character in the word string is in the letter_bank array
+    # (in the right quantities) i.e. if the user's guess contains 3 Y's and
+    # the hand only has 1, then it will return false
+    # we cannot use a set for duplicating the letter_bank variable
+    # because it only store unique elements! 
+    # will not be helpful with duplicates
+
+    word_is_in_letter_bank = True
+    string_letter_bank = ""
+
+    for letter in letter_bank: 
+        string_letter_bank += letter
+    
+    # check to see if there are more letters in the word than 
+    # in the letter_bank 
+    if len(word) > len(letter_bank): 
+        return False
+
+    for character in word:
+        character = str.upper(character)
+        if character in string_letter_bank: 
+            string_letter_bank = string_letter_bank.replace(character, '')
+        else: 
+            word_is_in_letter_bank = False
+            break
```

Comment:
Instead of setting a variable to `False` here, you can just return `False`. That way you don't even need the variable and the code is more efficient! If your code makes it out of the for loop, you know you can return `True`.